### ISSUE TITLE
[Performance] Downloading Files is Slow

### ISSUE THREAD

**Issue by lifewcody** [getinsomnia/insomnia#622]
- Insomnia Version: 5.12.0-beta.2
- Operating System: Windows 10 Pro 64bit

## Details

On Insomnia, downloading files takes a lot longer than in google chrome. Both tests are conducted on the same file.

Chrome:
Test 1: 4.92s
Test 2: 4.59s
Test 3: 4.55s
Avg. : ~4.69

Insomnia
Test 1: 13.4s (~30s)
Test 2: 12s (28.5s)
Test 3: 12s (28.6s)
Avg. : ~12.47 (~29.03)

The actual time Insomnia reports (with the HTTP code and the side) is the first number. The number in parenthesis is how long it showed loading on the screen.
Note: Around 25.8 seconds on screen is lags a little

**Comment by gschier** [comment]
How big of a file was this?

**Comment by lifewcody** [comment]
358MB

**Comment by gschier** [comment]
Since Insomnia is a debugging too there is probably no way it can be as fast as Chrome at downloading files, but I did realize the time discrepancy was caused by gzipping responses. This is why it appeared to take 30s but reported 12s. I've disabled gzip now because of this, which saves some time but it's still an order of magnitude slower.

The reason Insomnia is much slower is because every chunk of data that comes in gets sent from libcurl (C library) to Insomnia (Javascript). Large files can have thousands of chunks so these many _slow_ transfers add up.

To see actual performance (disregarding network) I downloaded a 500MB file from a local server. Using Curl, it took less than a second to download and write to the filesystem. Insomnia, it takes 14.4 seconds.
